### ISSUE TITLE
replaceWith() messes with entities

### ISSUE THREAD

**Issue by tuxor1337** [gawel/pyquery#97]
Just noted this:

    >>> d = PyQuery("<p/>&lt;b")
    >>> d("p").replaceWith("a")
    [<p>]
    >>> d.outerHtml()
    u'<div>a<b></b></div>'

So the `&lt;` entity is translated to `<` and suddenly there's a new `<b/>` element and stuff is getting really weird...

**Comment by knitori** [comment]
https://github.com/gawel/pyquery/blob/master/pyquery/pyquery.py#L1327

tag.tail is the [text](http://lxml.de/api/lxml.etree._Element-class.html#tail) after the tag, so this probably needs escaping. Line 1322 probably too.

**Comment by tuxor1337** [comment]
Yes, indeed, it seems like there is no option to access the tail without having etree parse all the entities. So, if we want to reuse the tail in html code, we have to re-escape it ...

    >>> div = lxml.etree.fromstring("<div><p/>&lt;b</div>")
    >>> [el.tail for el in div.xpath("p")]
    ['<b']

Unfortunately, up to now there is no part of pyquery that makes of any html-escape function. So we have to decide which one is the most elegant one in this case. Do we want to manually replace("<","&lt;").replace(">","&gt;") or do we want to import from another external package and use one of the following:

https://docs.python.org/2/library/xml.sax.utils.html#xml.sax.saxutils.escape
https://docs.python.org/2/library/cgi.html#cgi.escape

**Comment by gawel** [comment]
+1 for sax. cgi.escape is deprecated in 3.x
